pandas/core/reshape/reshape.py

```diff
@@ -291,18 +291,19 @@ def get_new_values(self, values, fill_value=None):
         # if our mask is all True, then we can use our existing dtype
         if mask_all:
             dtype = values.dtype
-            new_values = np.empty(result_shape, dtype=dtype)
```

Comment:
I think you can remove the 3 lines above this one too

Reply:
Indeed.